executor/executor.go

```diff
@@ -320,8 +334,17 @@ func Run(path string, env []string, emitter screwdriver.Emitter, build screwdriv
 
 		fReader := bufio.NewReader(f)
 
+		exportEnvCommand := []string{
+			"set -e",
+			"export PATH=$PATH:/opt/sd",
+			exportEnvCmd,
+		}
+
 		go func() {
 			runCode, rcErr := doRunCommand(guid, stepFilePath, emitter, f, fReader)
+			// run export file after step execution
+			shargs := strings.Join(exportEnvCommand, " && ")
+			f.Write([]byte(shargs))
```

Comment:
this could be done inside the case block for `stepAbort` so we can grab the exact state of env variables

Reply:
Tested with this change, it doesn't export the env var.
https://beta.cd.screwdriver.cd/pipelines/6512/builds/119164/steps/teardown-step1
https://beta.cd.screwdriver.cd/pipelines/6512/builds/119166/steps/teardown-step1

Reply:
@pritamstyz4ever checked build timeout doesn't carry the env variables to teardown steps, so this is needed.
only prob I see with f.Write is, it keeps printing this line `set -e && export PATH=$PATH:/opt/sd && tmpfile=/tmp/env_tmp; exportfile=/tmp/` at the end of each step, wondering how its not printing for your build.